src/librustc/infer/error_reporting/anon_anon_conflict.rs

```diff
@@ -0,0 +1,216 @@
+                                hir::ItemFn(ref fndecl, _, _, _, _, _) => {
+                                    fndecl
+                                        .inputs
+                                        .iter()
+                                        .filter_map(|arg| {
+                                            let mut nested_visitor = FindNestedTypeVisitor {
+                                                infcx: &self,
+                                                hir_map: &self.tcx.hir,
+                                                bound_region: *br,
+                                                found_type: None,
+                                            };
+                                            nested_visitor.visit_ty(&**arg);
+                                            if nested_visitor.found_type.is_some() {
+                                                return nested_visitor.found_type;
+                                            } else {
+                                                return None;
+                                            }
+                                        })
+                                        .next()
+                                }
+                                _ => None,
+                            }
+                        }
+                        _ => None,
+                    }
+                }
+                _ => None,
+            }
+        } else {
+            None
+        }
+    }
+
+    pub fn try_report_anon_anon_conflict(&self, error: &RegionResolutionError<'tcx>) -> bool {
+
+        let (span, sub, sup) = match *error {
+            ConcreteFailure(ref origin, sub, sup) => (origin.span(), sub, sup),
+            _ => return false, // inapplicable
+        };
+
+        // Determine whether the sub and sup consist of both anonymous (elided) regions.
+        let (ty1, ty2) = if self.is_anonymous_region(sup).is_some() &&
+                            self.is_anonymous_region(sub).is_some() {
+            let br1 = self.is_anonymous_region(sup).unwrap();
+            let br2 = self.is_anonymous_region(sub).unwrap();
+
+            if self.find_anon_type(sup, &br1).is_some() &&
+               self.find_anon_type(sub, &br2).is_some() {
+                (self.find_anon_type(sup, &br1).unwrap(), self.find_anon_type(sub, &br2).unwrap())
+            } else {
+                return false;
+            }
+        } else {
+            return false; // inapplicable
+        };
+
+        if let (Some(sup_arg), Some(sub_arg)) =
+            (self.find_arg_with_anonymous_region(sup, sup),
+             self.find_arg_with_anonymous_region(sub, sub)) {
+            let ((anon_arg1, _, _, _), (anon_arg2, _, _, _)) = (sup_arg, sub_arg);
+
+            let span_label_var1 = if let Some(simple_name) = anon_arg1.pat.simple_name() {
+                format!("from `{}`", simple_name)
+            } else {
+                format!("data flows here")
```

Comment:
In the case of a complex pattern, the error would not show any of the parameters but just `data flows here`. Not sure if that's what you meant.